rclcpp/include/rclcpp/mapped_ring_buffer.hpp

```diff
@@ -180,29 +225,52 @@ class MappedRingBuffer : public MappedRingBufferBase
    * It is up to the user to ensure the key is unique.
    * This method should not allocate memory.
    *
-   * After insertion, if a pair was replaced, then value will contain ownership
-   * of that displaced value. Otherwise it will be a nullptr.
+   * After insertion the value will be a nullptr.
+   * If a pair were replaced, its smart pointer is reseted.
    *
    * \param key the key associated with the value to be stored
    * \param value the value to store, and optionally the value displaced
    */
   bool
-  push_and_replace(uint64_t key, ElemUniquePtr & value)
+  push_and_replace(uint64_t key, const ConstElemSharedPtr & value)
   {
     std::lock_guard<std::mutex> lock(data_mutex_);
     bool did_replace = elements_[head_].in_use;
     elements_[head_].key = key;
-    elements_[head_].value.swap(value);
+    elements_[head_].unique_value.reset();
```

Comment:
I realize this was preexisting, but do you guys think it's worth saving a reference to `elements_[head_]` rather than accessing it with the `operator[]` each time? I assume the compiler is amazing and avoids any overhead here, but maybe it would be better than accessing it like this over and over when a reference should work just fine too.